sdk/resourcemanager/azure-resourcemanager-servicebus/src/main/java/com/azure/resourcemanager/servicebus/implementation/DurationSerializer.java

```diff
@@ -0,0 +1,121 @@
+        return module;
+    }
+
+    @Override
+    public void serialize(Duration duration, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
+        throws IOException {
+        jsonGenerator.writeString(DurationSerializer.toString(duration));
+    }
+
+    /**
+     * Convert to provided Duration to an ISO 8601 String with a days component.
+     * @param duration The Duration to convert.
+     * @return The String representation of the provided Duration.
+     */
+    public static String toString(Duration duration) {
+        String result = null;
+        if (duration != null) {
+            if (duration.isZero()) {
+                result = "PT0S";
+            } else {
+                final StringBuilder builder = new StringBuilder();
+
+                builder.append('P');
+
+                final long days = duration.toDays();
+                if (days > 0) {
+                    builder.append(days);
+                    builder.append('D');
+                    duration = duration.minusDays(days);
+                }
+
+                final long hours = duration.toHours();
+                if (hours > 0) {
+                    builder.append('T');
+                    builder.append(hours);
+                    builder.append('H');
+                    duration = duration.minusHours(hours);
+                }
+
+                final long minutes = duration.toMinutes();
+                if (minutes > 0) {
+                    if (hours == 0) {
+                        builder.append('T');
+                    }
+
+                    builder.append(minutes);
+                    builder.append('M');
+                    duration = duration.minusMinutes(minutes);
+                }
+
+                final long seconds = duration.getSeconds();
+                if (seconds > 0) {
+                    if (hours == 0 && minutes == 0) {
+                        builder.append('T');
+                    }
+
+                    builder.append(seconds);
+                    duration = duration.minusSeconds(seconds);
+                }
+
+                long nanoseconds = duration.getNano();
+                if (nanoseconds > 0) {
+                    if (hours == 0 && minutes == 0 && seconds == 0) {
+                        builder.append("T");
+                    }
+
+                    if (seconds == 0) {
+                        builder.append("0");
+                    }
+
+                    builder.append('.');
+
+                    long nanoMax = 100000000L;
+                    while (nanoseconds < nanoMax) {
+                        builder.append('0');
+                        nanoMax = nanoMax / 10;
+                    }
+
+                    // Remove trailing zeros.
+                    while (nanoseconds % 10 == 0) {
+                        nanoseconds /= 10;
+                    }
+                    builder.append(nanoseconds);
+                }
```

Comment:
Need a customized `DurationSerializer` to write with nanosecond precision. Else service would think the value being changed.

Rest of the code is same as the one in azure-core, which only write with millisecond precision.